pkg/stanza/operator/parser/jsonarray/parser_test.go

```diff
@@ -8,23 +8,30 @@ import (
 	"time"
 
 	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/featuregate"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/entry"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/testutil"
 )
 
 func newTestParser(t *testing.T) *Parser {
+	err := featuregate.GlobalRegistry().Set("logs.jsonParserArray", true)
+	require.Nil(t, err)
```

Comment:
Let's use [`testutil.SetFeatureGateForTest`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/29d1f4fb97d42f5ef09a3132570d64321a9c1449/internal/common/testutil/testutil.go#L130).

Can we also get the string value from `jsonArrayParserFeatureGate` or make it a constant?

```suggestion
	defer testutil.SetFeatureGateForTest(t, enableParser, true)
```

Reply:
Thanks for your suggestion. 
`SetFeatureGateForTest` resets the original state very carefully